packages/api/src/utils/psa.js

```diff
@@ -56,19 +61,192 @@ export const psaStatusesToDBStatuses = (statuses) => {
+  properties: {
+    name: { type: 'string', maxLength: 255 },
+    after: { type: 'string', format: 'date-time' },
+    before: { type: 'string', format: 'date-time' },
+    cid: { type: 'array', items: { type: 'string' } },
+    limit: { type: 'integer', minimum: 1, maximum: MAX_PIN_LISTING_LIMIT },
+    meta: { type: 'object' },
+    match: {
+      type: 'string',
+      enum: ['exact', 'iexact', 'ipartial', 'partial']
+    },
+    status: {
+      type: 'array',
+      items: {
+        type: 'string',
+        enum: ['queued', 'pinning', 'pinned', 'failed']
+      }
+    }
+  }
+})
+
+const postPinValidator = new Validator({
+  type: 'object',
+  required: ['cid'],
+  properties: {
+    cid: { type: 'string' },
+    name: { type: 'string', maxLength: 255 },
+    meta: { type: 'object', keys: { type: 'string' } },
+    origins: { type: 'array', items: { type: 'string' }, maxItems: 20 },
+    requestId: { type: 'string' }
+  }
+})
+
+/**
+ * Helper function to parse and validate payload for POST endpoint.
+ *
+ * @param {*} payload
+ * @returns parsedData
+ */
+export function transformAndValidate (payload) {
+  /** @type {*} */
+  const opts = {}
+  const {
+    cid,
+    name,
+    meta,
+    origins,
+    requestId
+  } = payload
+  let normalizedCid
+
+  // Validate CID.
+  if (cid) {
+    try {
+      normalizedCid = normalizeCid(cid)
+      opts.cid = cid
+    } catch (e) {
+      return {
+        error: new PSAErrorInvalidData(INVALID_CID),
+        data: undefined,
+        normalizeCid: undefined
+      }
+    }
+  }
+
+  if (name) opts.name = name
+  if (meta) opts.meta = meta
+  if (origins) opts.origins = origins
+  if (requestId) opts.requestId = requestId
+
+  const result = postPinValidator.validate(opts)
+
+  let data
+  let error
+
+  if (result.valid) {
+    data = opts
+  } else {
+    error = parseValidatorErrors(result.errors)
+  }
+
+  return { data, normalizedCid, error }
+}
+
+/**
+ * Helper function to parse and validate queryString for GET endpoint.
+ *
+ * @param {*} queryString
+ * @returns parsedData
+ */
+export function validateSearchParams (queryString) {
+  /** @type {*} */
+  const opts = {}
+  const {
+    after,
+    before,
+    cid,
+    limit,
+    match,
+    meta,
+    name,
+    status
+  } = queryString
+
+  // Validate CID(s).
+  if (cid) {
+    const cids = cid.split(',')
+    const sourceCids = []
+    for (const cid of cids) {
+      try {
+        normalizeCid(cid)
+        sourceCids.push(cid)
+      } catch (e) {
+        return {
+          error: new PSAErrorInvalidData(INVALID_CID),
+          data: undefined
+        }
+      }
+    }
+    opts.cid = sourceCids
+  }
+
+  if (limit) {
+    opts.limit = Number(limit)
+  } else {
+    opts.limit = DEFAULT_PIN_LISTING_LIMIT
+  }
+
+  if (name) opts.name = name
+  if (meta) opts.meta = meta
+  if (match) opts.match = match
+  if (before) opts.before = before
+  if (after) opts.after = after
+  if (status) opts.status = status.split(',')
```

Comment:
```suggestion
  opts.status = status.split(',')
```

Reply:
it will always exist at this stage or failed, right?

Reply:
At this stage it may or may not exist. Removing the conditional will throw `Cannot read properties of undefined (reading 'split')`